netmiko/huawei/huawei_smartax.py

```diff
@@ -131,3 +131,19 @@ def save_config(
         return super().save_config(
             cmd=cmd, confirm=confirm, confirm_response=confirm_response
         )
+
+    def cleanup(self, command: str = "quit") -> None:
+        """Gracefully exit the SSH session."""
+        super().cleanup(command=command)
+        start_time = time.time()
+        output: str = ""
+        # If after 30 seconds the device hasn't logged out, force it
+        while time.time() - start_time < 30:
+            output += self.read_channel()
+            if output.endswith("Are you sure to log out? (y/n)[n]:"):
+                self.write_channel("y" + self.RETURN)
+                output = ''
+                continue
+            if output.endswith("Configuration console exit, please retry to log on\n"):
+                break
+            time.sleep(0.01)
```

Comment:
I would probably break this into `new_output` and `output` and check for patterns in `new_output`. Using `endswith` sounds a bit dangerous to me as any extra text/newline will cause this to not work.

Technically since we don't return `output` or do anything with it, we could just capture `new_output` and not even keep a running `output`.